builder/builder.go

```diff
@@ -0,0 +1,68 @@
+package builder
+
+import (
+	"errors"
+
+	"github.com/NethermindEth/juno/blockchain"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/mempool"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/NethermindEth/juno/vm"
+)
+
+type Builder struct {
+	ownAddress felt.Felt
+
+	bc *blockchain.Blockchain
+	vm vm.VM
+
+	log utils.Logger
+}
+
+func New(ownAddr *felt.Felt, bc *blockchain.Blockchain, builderVM vm.VM, log utils.Logger) *Builder {
```

Comment:
I guess we'll probably want a private key as well (maybe in lieu of the address?). But it's too early to worry about that.